### PR TITLE
Add vulkan integer and float datatype support

### PR DESCRIPTION
**Pull request by YangleiZouIntel**
Add vulkan rt support to integer and float types.
Refactored buffer binding functions.

### CODE REVIEW

File: vendor/openvino/repo.bzl [hunk 1]
```diff
@@ -32,8 +32,8 @@ def openvino_workspace():
 
     http_archive(
         name = "openvino",
-        sha256 = "7b416d45765aea2cdd42edcdfc4e6ab4d7c4d37587530bf51a43b4c261e24a16",
-        strip_prefix = "dldt-2020.1",
+        sha256 = "0d641b158fff868ae7a969c02bf73462e335c15c40b42e3fe65ac6ea7cffea66",
+        strip_prefix = "openvino-2020.1",
         url = "https://github.com/opencv/dldt/archive/2020.1.zip",
```
Comment (YangleiZouIntel) [inline vendor/openvino/repo.bzl:37]: This is not related to the buffer binding. But the openvino zip file seems to have changed, I have to update the sha256 to be able to compile and pass CI. Please double check.

Comment (flaub) [reply vendor/openvino/repo.bzl:37]: Good catch, could you make this into a separate PR?

File: pmlc/conversion/gpu/gpu_to_vulkan.cc [hunk 2]
```diff
@@ -109,15 +126,61 @@ class ConvertGpuLaunchFuncToVulkanCalls
   }
 
   const char *getBufferBindingFunc(Type &elementType) {
```
Comment (flaub) [inline pmlc/conversion/gpu/gpu_to_vulkan.cc:128]: Don't pass `Type` by-reference, always use by-value semantics.

File: pmlc/conversion/gpu/gpu_to_vulkan.cc [hunk 3]
```diff
@@ -109,15 +126,61 @@ class ConvertGpuLaunchFuncToVulkanCalls
...
     if (elementType.isF32()) {
       return kBindBufferFloat32;
     }
+    if (elementType.isF64()) {
+      return kBindBufferFloat64;
+    }
     return nullptr;
   }
 
+  void getElementTypeSize(Type &elementType, uint32_t &size) {
```
Comment (flaub) [inline pmlc/conversion/gpu/gpu_to_vulkan.cc:156]: I'm not sure why we need this method. Isn't there a `type.getIntOrFloatBitWidth()` that you can use?

File: vendor/openvino/repo.bzl (hunk 1 above)
Comment (YangleiZouIntel) [reply vendor/openvino/repo.bzl:37]: Sure

File: pmlc/conversion/gpu/gpu_to_vulkan.cc (hunk 3 above)
Comment (YangleiZouIntel) [reply pmlc/conversion/gpu/gpu_to_vulkan.cc:156]: Thanks for pointing to this function, I will replace it by this function.

File: pmlc/conversion/gpu/gpu_to_vulkan.cc [hunk 4]
```diff
@@ -50,8 +50,20 @@ static constexpr const char *kCreateVulkanMemoryTransferAction =
...
+static constexpr const char *kBindBufferFloat64 = "bindBufferFloat64";
+
+// These functions are signless, meaning they apply to both signed and unsigned
+// integers
+static constexpr const char *kBindBufferInteger8 = "bindBufferInteger8";
+static constexpr const char *kBindBufferInteger16 = "bindBufferInteger16";
+static constexpr const char *kBindBufferInteger32 = "bindBufferInteger32";
+static constexpr const char *kBindBufferInteger64 = "bindBufferInteger64";
+
+static constexpr const int byteBits = 8;
```
Comment (flaub) [inline pmlc/conversion/gpu/gpu_to_vulkan.cc:66]: Use `kByteBits` for constant values.

File: pmlc/conversion/gpu/gpu_to_vulkan.cc [hunk 5]
```diff
@@ -210,14 +235,27 @@ ConvertGpuLaunchFuncToVulkanCalls::bindBuffers(Location loc, OpBuilder &builder,
...
     if (auto memRefType = buffer.getType().dyn_cast_or_null<MemRefType>()) {
+      auto shape = memRefType.getShape();
+      uint32_t numElement = 1;
+      for (auto dim : shape) {
+        numElement *= dim;
+      }
+
       auto elementType = memRefType.getElementType();
+      bufferElementTypes.push_back(elementType);
+      uint32_t elementTypeSize = elementType.getIntOrFloatBitWidth() / byteBits;
```
Comment (flaub) [inline pmlc/conversion/gpu/gpu_to_vulkan.cc:246]: Use `llvm::divideCeil` to account for `i1` types.

Comment (YangleiZouIntel) [reply pmlc/conversion/gpu/gpu_to_vulkan.cc]: It is a good point, thanks.

**Pull request merged**